p2p/proto/block.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+import "google/protobuf/timestamp.proto";
+
+// for now, we assume a small consensus, so this fits in 1M. Else, these will be repeated
+message Signatures {
+    BlockID id = 1;
+    repeated Signature signatures = 2; // can be more explicit here about the signature structure as this is not part of account abstraction
+}
+
+// Note: commitments may change to be for the previous blocks like comet/tendermint
+// hash of block header sent to L1
+message BlockHeader {
+    BlockID parent_block = 1;
+
+    google.protobuf.Timestamp time = 2; // TODO: see if this needs to be Felt252 or can be converted
+
+    Address sequencer_address = 3;
+
+    Merkle state_diffs = 4; //  By order of (contract, key), taking last in case of duplicates.
+                            // This means the proposer needs to sort after finishing the block (TBD: patricia? )
+
+    // State is optional and appears every X blocks for the last block. This is to support
+    // snapshot sync and also so that light nodes can sync on state without state diffs.
+    Merkle state = 5; // hash of contract and class patricia tries. Same as in L1. Later more trees will be included
```

Comment:
The Starknet decentralization proposals mention [checkpoint blocks](https://community.starknet.io/t/starknet-decentralized-protocol-v-checkpoints-for-fast-finality/6032). Is this `state` field what effectively define checkpoint blocks? Are quorum-certificates no longer going to be included? Thanks.

Reply:
@ittaysw Do you mind explaining the rationale behind not including the state root in each block again?

Reply:
@rianhughes , we're currently not going with checkpoints. 
@omerfirmak, nodes can keep track of state by state diffs, which they will need anyway. The state commitment if for light nodes, so opting to not waste the cycles on all blocks for this.

Reply:
So, will sequencers skip commitment calculation for those blocks that don't include it?

Reply:
Right

Reply:
Is it OK that we wont be able to generate merkle proofs for storage locations on these blocks?

Reply:
For snap sync, a node picks a pivot block and start fetching leaves of the state of that block. These leaves are verified against the state commitment of that particular pivot block. This requires both parties to agree on a state commitment. If the header doesnt include the commitment, agreeing on a commitment is not straight forward, if not impossible. So we are essentially forced to use only the blocks with commitments as pivot blocks.

As pivot block gets older, node may switch to a more recent block to use as pivot. Mainly because peers may decide to prune the state for that height and not be able to serve proofs/leaves for that block. 

Depending on how frequent are the blocks with commitments, it might make snap syncing impossible/inefficient. If the state for the pivot block a node is syncing against gets pruned thru out most of the network and no new blocks with commitments exist, we have no way to switch pivot to a more recent block.

Reply:
I didn't think state commitment would be rare. Something like every 10 blocks (so a light node has a frequency of 100 seconds if a block is 10 seconds).